storage/store.go

```diff
@@ -2506,6 +2511,8 @@ func (s *Store) processRaft() {
 			s.pendingRaftGroups.Unlock()
 			s.mu.Unlock()
 
+			s.processRaftMu.Lock()
```

Comment:
With the per-Replica mutex, you would hold on to that mutex before `s.mu.Unlock()` above, right?